arch/risc-v/src/common/supervisor/riscv_perform_syscall.c

```diff
@@ -46,6 +46,16 @@ void *riscv_perform_syscall(uintreg_t *regs)
     {
       (*running_task)->xcp.regs = regs;
     }
+  else
+    {
+      /* Update scheduler parameters *running_task == this_task() */
+
+      nxsched_resume_scheduler(*running_task);
+
+#ifdef CONFIG_ARCH_ADDRENV
+      addrenv_switch(*running_task);
```

Comment:
why call addrenv_switch here and line 80? could we merge two places into one?

Reply:
done